service/src/main/java/bio/terra/workspace/db/FolderDao.java

```diff
@@ -0,0 +1,254 @@
+package bio.terra.workspace.db;
+
+import bio.terra.common.db.ReadTransaction;
+import bio.terra.common.db.WriteTransaction;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.common.exception.MissingRequiredFieldException;
+import bio.terra.workspace.db.exception.DuplicateFolderDisplayNameException;
+import bio.terra.workspace.db.exception.DuplicateFolderIdException;
+import bio.terra.workspace.db.exception.FolderNotFoundException;
+import bio.terra.workspace.db.exception.WorkspaceNotFoundException;
+import bio.terra.workspace.service.folder.model.Folder;
+import com.google.common.collect.ImmutableList;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.dao.support.DataAccessUtils;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FolderDao {
+  private static final String DEFAULT_ROOT = "0";
```

Comment:
First, someone might want to create a folder called `0`.

Second, why do we need to set something? Can we say, if parentFolderId is blank, folder is top-level folder?

Reply:
i picked 0 because I want this root folder id to be a secret that only the Dao knows about because the root folder does not have a database entry. If it's a UUID, it's easier to pass it back up as a parent id all the way to the endpoint response. This way we can catch it during string to uuid conversion